readme.adoc

```diff
@@ -131,21 +131,24 @@ class hello implements Runnable {
 [source, java]
 ----
 ///usr/bin/env jbang "$0" "$@" ; exit $?
-//DEPS com.sun.net.httpserver:http:20070405
+//DEPS io.undertow:undertow-core:2.2.37.Final
 
-import com.sun.net.httpserver.*;
 import java.io.IOException;
-import java.net.InetSocketAddress;
+import io.undertow.Undertow;
+import io.undertow.server.HttpHandler;
+import io.undertow.server.HttpServerExchange;
+
+class WebServer implements HttpHandler {
+
+    @Override
+    public void handleRequest(HttpServerExchange exchange) throws Exception {
+        exchange.getResponseSender().send("Hello from JBang!");
+    }
 
-class WebServer {
     public static void main(String[] args) throws IOException {
-        HttpServer server = HttpServer.create(new InetSocketAddress(8000), 0);
-        server.createContext("/", exchange -> {
-            String response = "Hello from JBang!";
-            exchange.sendResponseHeaders(200, response.length());
-            exchange.getResponseBody().write(response.getBytes());
-            exchange.close();
-        });
+        Undertow server = Undertow.builder()
+            .addHttpListener(8000, "localhost")
+            .setHandler(new WebServer()).build();
```

Comment:
could simplify it to not require separeate method declaration, inheritance and the new constructor.

```suggestion
            .setHandler(exchange -> exchange.getResponseSender().send("Hello from JBang!")).build();
```